packages/utils/src/createId.ts

```diff
@@ -35,24 +35,19 @@ export function createId(prefix = ID_PREFIX): string {
  * @returns An accessor for the generated id.
  */
 export function createSlotId(
-  prefix?: string,
-  deps: Accessor<any>[] = []
-): Accessor<string | undefined> {
+  prefix?: string
+): [id: Accessor<string | undefined>, track: VoidFunction] {
   const id = createId(prefix);
 
   const [slotId, setSlotId] = createSignal<string | undefined>(id);
 
-  createRenderEffect(
-    on(deps, () => {
-      setSlotId(id);
+  const updateSlotId = () => setSlotId(document.getElementById(id) ? id : undefined);
+  queueMicrotask(updateSlotId);
```

Comment:
Looks like it always run once, so i think we are good for the case that didn't use a deps array.